wordpress-mobile/WordPress-Android
# Post Settings: Sharing PostModel instance with the activity

As discussed on Slack, this PR introduces a way to share the post object instance between the settings fragment and the activity. This greatly simplifies when and how the post should be saved as it's completely delegated to the activity now. Since we refresh the settings views when it'll be shown, this should also give us confidence that any changes made in the activity to the post instance will be reflected in the settings unless it's through a callback in the activity. If we find any cases where we update the post from a callback in the activity, we can always update that specific view. Note that I have not found any such case just yet.

This should be the last PR before we start the wrap-up process for the whole feature. This wrap-up process will include design review, full testing and possibly a code review.

To test:
* Aside from changing a few fields, going back to list and coming back to see if they are properly updated, I'd suggest not spending too much time on testing on this PR. As mentioned above, there will be a full testing phase and if we find any UI issues there (even if it's related to this PR) we'll open separate PRs for them. I'll try to document the testing scenarios and it'll be just easier to do the tests together.

Having said that please feel free to test the whole feature and let me know if you come across any issues. Thanks!

It'd be great if either @aforcier or @maxme can review this one.
